### ISSUE TITLE
Y|X, t training failed

### ISSUE THREAD

**Issue by kevinwhan** [gsbDBI/ds-wgan#10]
Hi,

I am trying to generate many several Y's simultaneously (some are binary and some are continuous) based on X, t. It tuened out that the training failed since the test error and training loss will both blow up to infinity. It works if I want to generate only continuous Y's.

I include both my data and the jupyter notebook file that I use based on your colab example. Could you kindly have a look? Thanks!
[example.zip](https://github.com/gsbDBI/ds-wgan/files/4201424/example.zip)

**Comment by evanmunro** [comment]
Thanks for opening an issue. To troubleshoot it often helps to start small. I suggest trying to generate just a couple of X first jointly. And once you get the hang of it you can add complexity, like generating all variables jointly and then generating a few conditionally. that should help you narrow down if there is a bug and if so where in the code it is. Unfortunately we can’t go through all your code and data ourselves from scratch without some troubleshooting first.

**Comment by kevinwhan** [comment]
Thanks for your quick reply! I have no difficulty in the first step for generating X|t. However, for generating Y|X, t, I found that if I want to generate both continuous and binary Y's together then I failed. There is no problem to generate only continuous Y's. Given this situation, do you suggest trying to generate Y's seperately (i.e. not generate all the outcomes variables simultaneously)? Thank you!

**Comment by evanmunro** [comment]
Does it work generating the binary Y variables only ?

**Comment by kevinwhan** [comment]
I want to generate 36 continuous Y's and 108 binary Y's. It works if I only generate 25 binary Y's or 36 continuous Y's with 10 binary Y's. If I want to generate more, it fails.

**Comment by evanmunro** [comment]
If it is not a specific X that is a problem then maybe the gradients are exploding. Try increasing the gradient penalty ?

**Comment by kevinwhan** [comment]
It works. Thank you!
